### DISCUSSION TITLE
Auto-updater should offer to include beta versions (but is disabled by default)

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3630] [Enhancement ideas (and their technical discussions)]
## Description

Users of GDevelop need to trust that software updates are well tested and reliable.
To make that happen, GDevelop needs more people to test new code before it is generally released.
Currently, users must watch for a beta version to be released, then download and install it manually.  Due to this manual process, not enough people are testing the beta versions.

## Solution suggested

Add a checkbox to "Preferences" under the "AutoDownload and install updates" that will allow beta versions to be included in the auto-updates.  This should be disabled by default, but it makes it very easy for users to participate in testing beta versions.

**Comment by 4ian** [comment]
Agree we should offer a way to easily download the beta version! 
Unsure exactly what electron-updater has to offer, but maybe there is a way to activate a "beta" track or something like this.

**Comment by arthuro555** [comment]
To be honest I'm wondering if we shouldn't try to switch to our own updater. The current one is based as far as I understand on the default electron-builder installer, based on NSIS, which many users experience problems with (like the installer getting stuck forever, updating unpinning GDevelop from the taskbar as all icons are removed and re-added, or refusing to install over a broken installation). As far as I understand too, updating currently more or less just "stupidly" downloads the latest release from GitHub and runs the installer in the background.

If we made our own installer/updater (like for example discord seem to do), that way we could actually fix the issues we have with the installer instead of having to say "sorry no can do" each time we get asked about it, update things more "intelligently", by downloading delta patches instead of the full thing, and only downloading what is needed (for example, download and apply separate delta patches for app.asar, each file of GDJS, libGD.wasm/.js, etc). 

I understand that this may require some effort that is "unwished" (we prefer to focus on working on GDevelop itself) but I think that the tradeoff is worth it, as a working installer and auto-updater is an expectation we want to meet, and while we do have "working" ones currently, they both repeatedly malfunctioned causing GDevelop to not meet those important expectations, so I think we need to find another solution than the built-in electron-builder ones.
